assembly/src/assembler/instruction/u32_ops.rs

```diff
@@ -310,48 +358,54 @@ pub fn u32popcnt(span: &mut SpanBuilder) -> Result<Option<CodeBlock>, AssemblyEr
     span.add_ops(ops)
 }
 
-/// Handles U32ADD, U32SUB, and U32MUL operations in wrapping, and overflowing modes, including
-/// handling of immediate parameters.
+/// Translates `u32clz` assembly instruction to VM operations. `u32clz` counts the number of
+/// leading zeros of the value using non-deterministic technique (i.e. it takes help of advice
+/// provider).
 ///
-/// Specifically handles these specific inputs per the spec.
-/// - Wrapping: does not check if the inputs are u32 values; overflow or underflow bits are
-///   discarded.
-/// - Overflowing: does not check if the inputs are u32 values; overflow or underflow bits are
-///   pushed onto the stack.
-fn handle_arithmetic_operation(
-    span: &mut SpanBuilder,
-    op: Operation,
-    op_mode: U32OpMode,
-    imm: Option<u32>,
-) -> Result<Option<CodeBlock>, AssemblyError> {
-    if let Some(imm) = imm {
-        push_u32_value(span, imm);
-    }
+/// This operation takes 27 VM cycles.
+pub fn u32clz(span: &mut SpanBuilder) -> Result<Option<CodeBlock>, AssemblyError> {
+    span.push_advice_injector(Clz);
+    span.push_op(AdvPop); // [clz, n, ...]
 
-    span.push_op(op);
+    calculate_clz(span)
```

Comment:
> I'm curious in general why we create new instructions as opposed to implementing those as standard library functions.

There isn't really a clear criteria for this, but here are some motivating factors:
1. Do other architectures have similar instructions at the assembly level? If so, might be good for us to have them as well.
2. Do we think these instructions are "core" enough so that people can use them without importing the standard library?

In the case of these instructions, I think it is worth to have them at the assembly level primarily because of point 1 above.

> 1. expose the new advice injectors (`Clz`, etc)
> 2. expose the `Expacc` operation
> 
> And I can't think of any reason why we wouldn't want to expose those?

I don't think there is any strong reason not to expose them - but that's probably for a different PR?